### ISSUE TITLE
How to know if delete was successful in a transaction?

### ISSUE THREAD

**Issue by HybridSolutions** [TanvirArjel/EFCore.GenericRepository#18]
Hi,

in a transaction, how would you delete an entity and verify the result before proceeding to another database operation?

For instance, with EF DbContext it's possible to get from the method _context.SaveChanges() an int with the number of records affected. That way, it's possible to commit or rollback. How do you do this with your library since all your methods do not return the value returned by the original dbcontext SaveChanges()? Why are you returning void?

This should be possible: 
`int affectedRows = await _repository.DeleteAsync(user);`

I suggest that all Delete and Update methods are updated to this:

```
public async Task<int> DeleteAsync<T>(T entity, CancellationToken cancellationToken = default)
            where T : class
        {
            if (entity == null)
            {
                throw new ArgumentNullException(nameof(entity));
            }

            _dbContext.Set<T>().Remove(entity);
            return await _dbContext.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
        }
```

```
public async Task<int> UpdateAsync<T>(IEnumerable<T> entities, CancellationToken cancellationToken = default)
            where T : class
        {
            if (entities == null)
            {
                throw new ArgumentNullException(nameof(entities));
            }

            _dbContext.Set<T>().UpdateRange(entities);
            return await _dbContext.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
        }
```

This way they will return the int returned by _dbContext.SaveChangesAsync().

Thank you

**Comment by HybridSolutions** [comment]
> @HybridSolutions Thank you so much for raising the issue. I shall take a look into it and come back to you soon.

Meanwhile I'm using the project source with the changes I mentioned already implemented and is working great. Hope you can add those changes too to use the package reference instead :)

**Comment by TanvirArjel** [comment]
@HybridSolutions This has been fixed and will be available in the next release.

**Comment by TanvirArjel** [comment]
Fix is now available in version 5.8.1.
